modules/govcms_ckan_media/includes/govcms_ckan_media.field_config.inc

```diff
@@ -170,16 +245,50 @@ function govcms_ckan_media_field_widget_form(&$form, &$form_state, $field, $inst
 }
 
 /**
- * Ajax callback for changing visualisations.
+ * Change the dataset url submit handler.
+ */
+function _govcms_ckan_media_field_widget_form_change_dataset_url($form, &$form_state) {
+  // Only proceed if explicitly set.
+  if (!$form_state['values']['dataset_url_change']) {
```

Comment:
@gollyg This submit handler gets added alongside the element so it should always exist (Line 163 & 203). That said I might cast to bool to make it more obvious.